### PR TITLE
Update the Rust toolchain to version 1.80.0

### PR DESCRIPTION
**Pull request by antklein**
Update to a newer Rust toolchain in order to incorporate improvements with newer versions.
For example, an improvement on the size optimization of output rust binaries.

### CODE REVIEW

**Review by makubacki — APPROVED** [review]
I'm going to approve assuming the testing we discussed looks good.

**Comment by antklein** [comment]
Validation still under test. Do not merge until testing is complete.

**Review by apop5 — APPROVED** [review]

**Pull request merged**
